lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,164 @@
+from ansible.module_utils._text import to_native
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.release import __version__
+
+try:
+    import docker
+    HAS_DOCKER = True
+except ImportError:
+    HAS_DOCKER = False
+
+
+class InventoryModule(BaseInventoryPlugin):
+    ''' Host inventory parser for ansible using Docker swarm as source. '''
+
+    NAME = 'docker_swarm'
+
+    def _get_tls_config(self, **kwargs):
+        try:
+            tls_config = docker.tls.TLSConfig(**kwargs)
+            return tls_config
+        except Exception as e:
+            raise AnsibleError('Unable to setup TLS, this was the original exception: %s' % to_native(e))
+
+    def _get_tls_connect_params(self):
+        if self.get_option('tls') and self.get_option('cert_path') and self.get_option('key_path'):
+            # TLS with certs and no host verification
+            tls_config = self._get_tls_config(client_cert=(self.get_option('cert_path'), self.get_option('key_path')),
+                                              verify=False)
+            return tls_config
+
+        if self.get_option('tls'):
+            # TLS with no certs and not host verification
+            tls_config = self._get_tls_config(verify=False)
+            return tls_config
+
+        if self.get_option('tls_verify') and self.get_option('cert_path') and self.get_option('key_path'):
+            # TLS with certs and host verification
+            if self.get_option('cacert_path'):
+                tls_config = self._get_tls_config(client_cert=(self.get_option('cert_path'), self.get_option('key_path')),
+                                                  ca_cert=self.get_option('cacert_path'),
+                                                  verify=True,
+                                                  assert_hostname=self.get_option('tls_hostname'))
+            else:
+                tls_config = self._get_tls_config(client_cert=(self.get_option('cert_path'), self.get_option('key_path')),
+                                                  verify=True,
+                                                  assert_hostname=self.get_option('tls_hostname'))
+
+            return tls_config
+
+        if self.get_option('tls_verify') and self.get_option('cacert_path'):
+            # TLS with cacert only
+            tls_config = self._get_tls_config(ca_cert=self.get_option('cacert_path'),
+                                              assert_hostname=self.get_option('tls_hostname'),
+                                              verify=True)
+            return tls_config
+
+        if self.get_option('tls_verify'):
+            # TLS with verify and no certs
+            tls_config = self._get_tls_config(verify=True,
+                                              assert_hostname=self.get_option('tls_hostname'),
+                                              ssl_version=self.get_option('ssl_version'))
```

Comment:
@WojciechowskiPiotr shouldn't `ssl_version` also be supplied in the other cases?

Reply:
If the version is not provided library fallback to TLSv1.2 or TLSv1.0 if very very old SSL library is installed on the host.

Reply:
Jip...forgot to take this one out too.